email/src/main/java/edu/jhuapl/dorset/demos/EmailManager.java

```diff
@@ -0,0 +1,308 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.jhuapl.dorset.demos;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import javax.mail.Flags;
+import javax.mail.Folder;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Multipart;
+import javax.mail.Part;
+import javax.mail.Session;
+import javax.mail.Store;
+import javax.mail.Transport;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigValue;
+
+public class EmailManager {
+    
+    private static final Logger logger = LoggerFactory.getLogger(EmailManager.class);
+    
+    private static final String USERNAME_KEY = "username";
+    private static final String PASSWORD_KEY = "password";
+    private static final String MAIL_STORE_TYPE_KEY = "mailStoreType";
+    private static final String HOST_KEY = "host";
+    private static final String FROM_KEY = "from";
+
+    private String user;
+    private String password;
+    private String mailStoreType;
+    private String host;
+    private String from;
+    
+    private Session session;
+    private Store store;
+    private Folder inboxFolder;
+    private Folder errorFolder;
+    private Folder completeFolder;
+    private static final String SMTP = "smtp";
+    private static final String ENCAPSULATED = "message/rfc822";
+    private static final String TEXT_PLAIN = "text/plain";
+    private static final String MULTIPART = "multipart/*";
+    
+    /**
+    * EmailManager Constructor.
+    * 
+    * The close method must be called on every EmailManager object before exiting an application
+    * 
+    * @param config  Configuration object that stores mail server information, username, and password
+    * @throws MessagingException   if connection cannot be established
+    */
+    public EmailManager(Config config) throws MessagingException {
+        
+        user = config.getString(USERNAME_KEY);
+        password = config.getString(PASSWORD_KEY);
+        mailStoreType = config.getString(MAIL_STORE_TYPE_KEY);
+        host = config.getString(HOST_KEY); 
+        from = config.getString(FROM_KEY);
+        Properties prop = extractProperties(config);
+        session = Session.getDefaultInstance(prop);
+        store = session.getStore(mailStoreType);
+        try {
+            store.connect(host, user, password);
+        } catch (MessagingException e) {
+            throw new MessagingException("Failed to set up imap connection", e);
+        }
+        try {
+            initFolders();
+            inboxFolder.open(Folder.READ_WRITE);
+        } catch (MessagingException e) {
+            throw new MessagingException("Failed to initialize and open folder");
```

Comment:
Here you should close the store since the connect worked but you're short circuiting the constructor because you couldn't open the folders.